internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go

```diff
@@ -220,21 +224,83 @@ func (h *PolicyChangeHandler) handleFleetServerHosts(ctx context.Context, c *con
 	return nil
 }
 
-func clientEqual(k1 remote.Config, k2 remote.Config) bool {
-	if k1.Protocol != k2.Protocol {
+// applyConfigWithPrecedence applies Proxy and TLS configs, but ignores empty or
+// absent ones.
+// That way a proxy or TLS config set during install/enroll using cli flags
+// won't be overridden by an absent or empty proxy from fleet-server.
+// However, if there is a proxy or TLS config sent by fleet-server, it'll take
+// precedence. Therefore, it's not possible to remove a proxy or TLS config once
+// it's set.
+func (h *PolicyChangeHandler) applyConfigWithPrecedence(cfg remote.Config) {
+	defaultcfg := configuration.DefaultFleetAgentConfig()
+
+	if cfg.Protocol != defaultcfg.Client.Protocol ||
+		cfg.Host != defaultcfg.Client.Host ||
+		!slices.Equal(cfg.Hosts, defaultcfg.Client.Hosts) ||
+		cfg.Path != defaultcfg.Client.Path {
+		h.config.Fleet.Client.Protocol = cfg.Protocol
+		h.config.Fleet.Client.Path = cfg.Path
+		h.config.Fleet.Client.Host = cfg.Host
+		h.config.Fleet.Client.Hosts = cfg.Hosts
+	}
+
+	if cfg.Transport.Proxy.URL == nil ||
+		cfg.Transport.Proxy.URL.String() == "" {
+		h.log.Debugw("proxy from fleet is empty or null, the proxy will not be changed")
+	} else {
+		if h.config.Fleet.Client.Transport.Proxy.URL != nil {
+			h.log.Debugw("received proxy from fleet, applying it",
+				"old_proxy_url", h.config.Fleet.Client.Transport.Proxy.URL.String(),
+				"new_proxy_url", cfg.Transport.Proxy.URL.String())
+		} else {
+			h.log.Debugw("received proxy from fleet, applying it",
+				"old_proxy_url", "nil",
```

Comment:
can be simplified with string variable `oldProxy = h.config.Fleet.Client.Transport.Proxy.URL == nil ? h.config.Fleet.Client.Transport.Proxy.URL.String() : "nil"` 
c# syntax but you get the idea